search/search_repository_blackbox_test.go

```diff
@@ -58,102 +61,213 @@ func (s *searchRepositoryBlackboxTest) getTestFixture() *tf.TestFixture {
+			// then
+			require.Nil(t, err)
+			require.Equal(t, uint64(1), count)
+			assert.Equal(t, wi2.ID, res[0].ID)
+			assert.Equal(t, res[0].Fields["system.order"], wi2.Fields["system.order"])
+		})
+
+		t.Run("type base", func(t *testing.T) {
+			// given
+			fxt := s.getTestFixture()
+			wi1 := fxt.WorkItems[0]
+			wi2 := fxt.WorkItems[1]
+			// when
+			res, count, err := s.searchRepo.SearchFullText(context.Background(), "TestRestrictByType type:"+fxt.WorkItemTypeByName("base").ID.String(), nil, nil, nil)
+			// then
+			require.Nil(t, err)
+			require.Equal(t, uint64(2), count)
+			assert.Equal(t, res[0].Fields["system.order"], wi2.Fields["system.order"])
+			assert.Equal(t, res[1].Fields["system.order"], wi1.Fields["system.order"])
+		})
+
+		t.Run("types sub1+sub2", func(t *testing.T) {
+			// given
+			fxt := s.getTestFixture()
+			wi1 := fxt.WorkItems[0]
+			wi2 := fxt.WorkItems[1]
+			// when
+			res, count, err := s.searchRepo.SearchFullText(context.Background(), "TestRestrictByType type:"+fxt.WorkItemTypeByName("sub2").ID.String()+" type:"+fxt.WorkItemTypeByName("sub1").ID.String(), nil, nil, nil)
+			// then
+			require.Nil(t, err)
+			assert.Equal(t, uint64(2), count)
+			assert.Equal(t, res[0].Fields["system.order"], wi2.Fields["system.order"])
+			assert.Equal(t, res[1].Fields["system.order"], wi1.Fields["system.order"])
+		})
+
+		t.Run("types base+sub1", func(t *testing.T) {
+			// given
+			fxt := s.getTestFixture()
+			wi1 := fxt.WorkItems[0]
+			wi2 := fxt.WorkItems[1]
+			// when
+			res, count, err := s.searchRepo.SearchFullText(context.Background(), "TestRestrictByType type:"+fxt.WorkItemTypeByName("base").ID.String()+" type:"+fxt.WorkItemTypeByName("sub1").ID.String(), nil, nil, nil)
+			// then
+			require.Nil(t, err)
+			assert.Equal(t, uint64(2), count)
+			assert.Equal(t, res[0].Fields["system.order"], wi2.Fields["system.order"])
+			assert.Equal(t, res[1].Fields["system.order"], wi1.Fields["system.order"])
+		})
+	})
+
+	s.T().Run("Filter with limits", func(t *testing.T) {
+
+		t.Run("none", func(t *testing.T) {
+			// given
+			fxt := s.getTestFixture()
+			// when
+			filter := fmt.Sprintf(`{"$AND": [{"space": "%s"}]}`, fxt.Spaces[0].ID)
+			res, count, err := s.searchRepo.Filter(context.Background(), filter, nil, nil, nil)
+			// when
+			require.Nil(t, err)
+			assert.Equal(t, uint64(2), count)
+			assert.Equal(t, 2, len(res))
+		})
+
+		t.Run("Filter with offset", func(t *testing.T) {
+			// given
+			fxt := s.getTestFixture()
+			// when
+			filter := fmt.Sprintf(`{"$AND": [{"space": "%s"}]}`, fxt.Spaces[0].ID)
+			start := 3
+			res, count, err := s.searchRepo.Filter(context.Background(), filter, nil, &start, nil)
+			// then
+			require.Nil(t, err)
+			assert.Equal(t, uint64(2), count)
+			assert.Equal(t, 0, len(res))
+		})
+
+		t.Run("Filter with limit", func(t *testing.T) {
+			// given
+			fxt := s.getTestFixture()
```

Comment:
Shouldn't this be enough here? `fxt := tf.NewTestFixture(s.DB(), t, tf.WorkItemLinks(3))` . All you want is to check that even if more than one exists, only one is returned (due to the limit), right?

Reply:
hum, right, I could avoid the call to `getTestFixture()` here, indeed